integration_test/index_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationtest
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"sigs.k8s.io/krew/pkg/constants"
+)
+
+func TestKrewIndexAdd(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.WithEnv(constants.EnableMultiIndexSwitch, 1).WithIndex()
+	if err := test.Krew("index", "add").Run(); err == nil {
+		t.Fatal("expected index add with no args to fail")
+	}
+	if err := test.Krew("index", "add", "foo", "https://invalid").Run(); err == nil {
+		t.Fatal("expected index add with invalid URL to fail")
+	}
+	if err := test.Krew("index", "add", "../../usr/bin", constants.IndexURI); err == nil {
+		t.Fatal("expected index add with path characters to fail")
+	}
```

Comment:
This case is also the case of an invalid  URL. File paths are perfectly valid repo locations for git and it will clone happily if the source exists (it's dangerous though, because both repos will share their blob store).

So I'd recommend to remove this case.

Reply:
I added this one for testing if the name of the index contains those path characters. It fails at a different point than when the provided URL is invalid. The error message is probably not worded super well since its pretty similar to the one with the invalid URL, I can update that to mention that the path characters are invalid in the name.

Reply:
Oh.. you're right, my bad. So what about extracting the name sanitizer logic into  its own function and test that instead? In the best case, we can reuse a k8s util anyway and we don't need to test anything.

Reply:
Ahmet commented the same and I have some changes which I have yet to push that moves the logic to its own function. I want to address all the comments before pushing the changes, so I'll look into any existing k8s name validation util. I'll also remove this test since we'll have that logic elsewhere.

Reply:
We should still test an invalid path case in integration_test, since it's CLI behavior. I'm not sure if `indexoperations` methods should be aware of security checks.